### ISSUE TITLE
maximum recursion depth exceeded error in `monkey_patch`

### ISSUE THREAD

**Issue by lobsterkatie** [eventlet/eventlet#660]
Hello!

[This PR](https://github.com/eventlet/eventlet/pull/654), merged yesterday, removed a version check on Python, and now running under 2.7 (which we do as part of our test suite) throws this error:

```
_______ ERROR at setup of test_transport_works[eventlet-True-close-True] _______
tests/conftest.py:292: in maybe_monkeypatched_threading
    eventlet.monkey_patch()
.tox/py2.7/lib/python2.7/site-packages/eventlet/patcher.py:316: in monkey_patch
    _os = original('os')
.tox/py2.7/lib/python2.7/site-packages/eventlet/patcher.py:201: in original
    real_mod = __import__(modname, {}, {}, modname.split('.')[:-1])
.tox/py2.7/lib/python2.7/fnmatch.py:40: in fnmatch
    import os
.tox/py2.7/lib/python2.7/fnmatch.py:40: in fnmatch
    import os
E   RuntimeError: maximum recursion depth exceeded in __instancecheck__
!!! Recursion detected (same locals & position)
```

That version check was added [here](https://github.com/eventlet/eventlet/commit/115103d5608cbe8f15df10e27eba1644f5364e95). Is it incompatible with the change made yesterday, or could it be added back in?

Thanks!

**Comment by temoto** [comment]
Fixed in 0.29.1

**Comment by lobsterkatie** [comment]
Thanks for the quick turn-around!
